src/test/java/com/conveyal/r5/streets/StreetLayerTest.java

```diff
@@ -124,6 +125,55 @@ public void testSplits() throws Exception {
         //streetLayer.edgeStore.dump();
     }
 
+    /**
+     * Test if edge length and geometry are properly preserved in splitting
+     */
+     @Test
+    public void testSplitsForStops () {
+         OSM osm = new OSM(null);
+         osm.intersectionDetection = true;
+
+         // One edge from Oakland, CA
+         osm.readFromUrl(StreetLayerTest.class.getResource("snake-rd.pbf").toString());
+         // Two coordinates near the edge
+         Coordinate[] stopCoordinates = new Coordinate[]{
+            new Coordinate(-122.206863, 37.825161),
+            new Coordinate(-122.206751, 37.826258)
+         };
+         StreetLayer streetLayer = new StreetLayer(TNBuilderConfig.defaultConfig());
+         streetLayer.loadFromOsm(osm, false, true);
+         osm.close();
+         //This is needed for inserting new vertices around coordinates
+         streetLayer.indexStreets();
+
+        long osmId = streetLayer.edgeStore.getCursor(0).getOSMID();
+        int originalLength = streetLayer.edgeStore.getCursor(0).getLengthMm();
+        Coordinate[] originalGeometry = streetLayer.edgeStore.getCursor(0).getGeometry().getCoordinates();
+
+         for (Coordinate stop : stopCoordinates) {
+            streetLayer.createAndLinkVertex(stop.y, stop.x);
+         }
+
+         int accumulatedLength = 0;
+         int originalGeometryIndex = 1;
+
+         EdgeStore.Edge edge = streetLayer.edgeStore.getCursor();
+
+         while(edge.advance()) {
```

Comment:
The first time advance() is called, the edge cursor is pointing at edge 0, so the same edge that was split, and from which the OSM ID and length were read. So I believe this test would not detect several failure modes, e.g. if the geometry split process completely failed or was a no-op, or if the split was non-destructive, did not temporarily delete the original edge, and did not set geometries on any of the resulting split edges.

To detect a wider range of failure modes, we should probably detect that each call to streetLayer.createAndLinkVertex returns a non-negative vertex ID, that the edge store has the expected number of edges after the operation, that each of those edges begins and ends at the expected vertex index, and that the original intermediate coordinates are split across all three resulting edges (rather than being concentrated in any one edge, i.e. that each new edge contributes some points to the comparison with the original geometry).

I know you need to make use of these changes ASAP and don't want to delay the merge too much, so I'm going to merge this one and take a shot at refining the test myself in another pull request.

Reply:
I'm working on the test now. I see it makes an assumption about the order of the generated edges, although order is not guaranteed by the contract of the splitting method. I'm making it a bit more tolerant of implementation changes and the addition of more split points to the test.

Reply:
I have revised the test over in PR #525, which now contains the changes in this PR via a merge to dev.